src/main/java/com/aws/greengrass/logmanager/util/FileHashGenerator.java

```diff
@@ -0,0 +1,63 @@
+package com.aws.greengrass.logmanager.util;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.logmanager.LogManagerService;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+import static com.aws.greengrass.util.Digest.calculate;
+
+public final class FileHashGenerator {
```

Comment:
It would be nice if we had a wrapper around the File class instead of having this `Generator` class which doesn't have much meaning. Maybe we have

```java
public class LogFile extends File {
     //... call super and so on

     getHash() <- your method

}

```